pkg/daemon/types/types.go

```diff
@@ -19,13 +18,20 @@ type BuildTimeInfo struct {
 type DaemonInfo struct {
 	ID      string        `json:"id"`
 	Version BuildTimeInfo `json:"version"`
-	State   string        `json:"state"`
+	State   DaemonState   `json:"state"`
 }
 
-func (info *DaemonInfo) Running() bool {
-	// Due to history reason, the daemon state returned by nydusd sock API has
-	// been changed from `Running` to `RUNNING`, so keeps compatibility here.
-	return strings.ToUpper(info.State) == "RUNNING"
+type DaemonState string
+
+const (
+	DaemonStateUnknown DaemonState = "UNKNOWN"
+	DaemonStateInit    DaemonState = "INIT"
+	DaemonStateReady   DaemonState = "READY"
+	DaemonStateRunning DaemonState = "RUNNING"
+)
+
+func (info *DaemonInfo) DaemonState() DaemonState {
```

Comment:
Now can we remove the method or make `state` filed private?

Reply:
We'd better keep it as since the daemon package will be exported as nydusd daemon library to be imported by other projects.

Reply:
Oh. yes. we already have State() method. Let's make it private.